reach4help/reach4help
# Only allow requests originating from Cloudflare

Description
-----------
This PR implements an IP filtering system to block any request that doesn't originate from a Cloudflare IP. The feature can be disabled when developing by setting `ENVIRONMENT` to `dev` in the `.env` file.

Disclaimer: I'm not a nodejs dev at all, and this is my first contact with nestjs, be indulgent please 😅. This solution feels a bit overkill compared to the goal, but from what I understood from the nestjs documentation this seems to be the way to go. If you have any idea of how I could make this feature simpler, please leave a comment, I'll be happy to make changes. 

Motivation
----------
To fully benefit from Coudflare's protection we need to ensure all the requests we handle have been proxied through them.

Testing Guidelines
------------------
Locally:
* in the `.env` file, set the `ENVIRONMENT` variable to anything else than `dev`
* make a request to the API
* you should get a 403 error

Release Checklist
-----------------


Security Checklist
-------------------

This PR creates, modifies, or deletes:




Additional Notes
----------------
closes #198
